src/main.js

```diff
@@ -0,0 +1,110 @@
+        launchContext: {
+            launchOptions: {
+                waitUntil: 'networkidle0',
+            },
+            useChrome: true,
+            stealth: false,
+        },
+        handlePageFunction: async ({ request, page }) => {
+            const { userData, url } = request;
+            let data = [];
+            log.info(`Processing ${url}...`);
+            if (userData.type === PAGE_TYPES.INDEX) {
+                // Index page
+                // Set "Results per page" select field to show all universities
+                await page.click('.sort_by_dropdown');
+                await Apify.utils.sleep(5000);
+                await page.click('.sort_by_dropdown .dropdown');
+                await Apify.utils.sleep(7000);
+                await page.click('.sort_by_dropdown div[data-value="100"]');
+                await Apify.utils.sleep(6000);
+                // Set country filter
+                if (country !== 'All countries') {
+                    log.info(`Setting desired country...`);
+                    await page.select('.country-select-realdiv #country-select', country.toLowerCase());
+                    await Apify.utils.sleep(6000);
+                }
+                const numberOfResults = await page.$eval('#_totalcountresults', (el) => el.textContent);
+                log.info(`Found ${numberOfResults} results in total.`);
+
+                if (numberOfResults > maxItems) {
+                    log.info(`Will be saved: ${maxItems} items.`);
+                }
+                // COUNTER OF ITEMS TO SAVE;
+                let itemsCounter = 0;
+
+                if (mode === MODE.VISIT_DETAIL) {
+                    const detailUrls = await page.$$eval('.uni-link', (links) => links
+                        .map((link) => `https://www.topuniversities.com/${link.getAttribute('href')}`));
+
+                    for (const req of detailUrls) {
+                        if (!(itemsCounter >= maxItems)) {
+                            await requestQueue.addRequest(
+                                {
+                                    url: req,
+                                    userData: {
+                                        type: PAGE_TYPES.DETAIL,
+                                    },
+                                },
+                            );
+                        }
+                        itemsCounter += 1;
+                    }
+                } else {
+                    // Basic mode
+                    data = await page.$$eval('.ind', pageFunction);
+                    for (const item of data) {
+                        if (!(itemsCounter >= maxItems)) {
+                            await Apify.pushData(item);
+                            itemsCounter += 1;
+                        }
+                    }
+                }
+            } else {
+                // University detail page
+                data = await page.$$eval('body', detailPageFunction);
```

Comment:
this is returning an array, the output is being `{ "0": {} }`. you can split the details when doing pushData (below)